dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java

```diff
@@ -1324,5 +1329,12 @@ public boolean canChangePassword(Context context, EPerson ePerson, String curren
     public boolean areSpecialGroupsApplicable(Context context, HttpServletRequest request) {
         return true;
     }
+
+    public String getEmailAcceptedOrNull(String email) {
+        if (email == null || email.isEmpty() || email.matches(".*\\s+.*")) { // no whitespaces in mail
```

Comment:
/\s/.test(email); ?

Reply:
isn't `/\s/.test(email)`  better?

Reply:
Maybe yes, but I don't understand that code. What is .test(email)? Some method?

Reply:
regex match vs search

Reply:
The code `/\\s/.test(email)` is written in JavaScript and it checks if the given email string contains any whitespace character. 

In Java it looks like this: `Pattern.compile("\\s").matcher(email).find()` 
I changed it to that Java Pattern condition.